src/sentry/deletions/defaults/notificationmessage.py

```diff
@@ -0,0 +1,17 @@
+from ..base import ModelDeletionTask, ModelRelation
+
+
+class NotificationMessageDeletionTask(ModelDeletionTask):
+    def get_child_relations(self, instance):
+        from sentry.incidents.models import AlertRuleTriggerAction
+
+        return [
+            ModelRelation(AlertRuleTriggerAction, {"notificationmessage_id": instance.id}),
+        ]
+
+    def mark_deletion_in_progress(self, instance_list):
+        from sentry.constants import ObjectStatus
+
+        for instance in instance_list:
+            if instance.status != ObjectStatus.PENDING_DELETION:
+                instance.update(status=ObjectStatus.PENDING_DELETION)
```

Comment:
This actually needs to be the other way around - this table is a child table of `AlertRuleTriggerAction`, so you need a `AlertRuleTriggerActionTask`.

You should also add a test for this similar to https://github.com/getsentry/sentry/blob/33f6772e7f8ad68bbd0035e2121a0ccdf0177694/tests/sentry/deletions/test_rule.py#L12

Reply:
Ohhh, woops, my bad. That totally makes sense. When an AlertRuleTriggerAction gets deleted, we need to make sure it doesn't cascade and time out, and of course it will call it's own deletion handler, not it's children.